src/Moq/ProxyFactories/CastleProxyFactory.cs

```diff
@@ -101,6 +101,15 @@ public void Intercept(Castle.DynamicProxy.IInvocation underlying)
 					this.interceptor.Intercept(invocation);
 					underlying.ReturnValue = invocation.ReturnValue;
 				}
+				catch (MockException)
+				{
+					throw;
+				}
+				catch (Exception ex)
```

Comment:
Is there a better way to intercept exceptions (I'm not really familiar with *CastleProxy*)?

Reply:
I thought the `Expcetion`property is for exceptions thrown by the method. `MockException` is not raised as part of the invocation, but before or after. I can't tell which is more sound: Storing MockException in `Exception` or not. Either way, the behavior should be covered by tests (which it is not yet).

Can you just tell me which way to go here?

Reply:
I changed the behavior so that `MockException` is stored in the `Exception` property and added a test in a064c982. Just tell me, if you find that behavior appropriate or not.